### PR TITLE
ENH: Record the active branch of a subdataset in the parent

### PR DESCRIPTION
**Pull request by mih**
This brings us closer to gh-1424

- [x] do not record the name of adjusted branches: https://github.com/datalad/datalad/pull/3817/checks?check_run_id=267367157#step:8:280
- [x] figure out why `git annex sync` blows up on windows (was https://github.com/datalad/datalad/issues/3824 and is fixed)
- [x] proposed handling involves a commit, which invalidates a bunch of test expectations.

### CODE REVIEW

**Comment by mih** [comment]
Windows test failure is somewhat related to #3818

**Comment by mih** [comment]
Rebased on master and gepushed.

**Comment by mih** [comment]
Remaining test failures are due to the adjusted branch carrying an additional commit after an `annex sync` despite `--no-commit`. This is a problem for any test that verifies the nature of "the last commit". In particular `diff` and `run` tests are affected.

Here is what a dataset like this feels like:

image

Unsure about the best way to proceed. Given that the branch info would be useful to have, we could

- adjust all tests to make them aware of this situation
- not run these tests

**Comment by kyleam** [comment]
> Given that the branch info would be useful to have, we could
> 
> * adjust all tests to make them aware of this situation (preferred)
> * not run these tests, with adjusted branches

I agree that the latter is the way to go.  I'll look at adjusting at least some of those tests today.

**Comment by kyleam** [comment]
I've pushed a fix for the easier cases.  (I'll update the check list once I verify that the pass.)  I think the remaining ones start to touch on deeper issues related to our code base not being designed to work with adjusted branches.  Moreover, I view the explicit sync call in `save` as just exposing existing issues with our handling, rather than creating new issues.  It probably makes sense to mark these as known failures and deal with them when we try to approach these larger issues (e.g., gh-3818).

**Comment by mih** [comment]
Thanks @kyleam for your assessment! I'll try implementing this missing pieces.

File: datalad/core/local/tests/test_diff.py [hunk 1]
```diff
@@ -84,7 +84,7 @@ def test_repo_diff(path, norepo):
     create_tree(ds.path, {'new': 'empty'})
     ds.save(to_git=True)
     assert_repo_status(ds.path)
-    eq_(ds.repo.diff(fr=fr_base + '~1', to=to),
+    eq_(ds.repo.diff(fr=fr_base + '~1', to=fr_base),
```
Comment (kyleam) [inline datalad/core/local/tests/test_diff.py]: Sorry about that.  Thanks for fixing my mistake.

**Comment by mih** [comment]
I am hopeful that this could be it!

**Comment by mih** [comment]
THIS IS IT!!

**Pull request merged**